RELEASE_CHECKLIST.md

```diff
@@ -0,0 +1,20 @@
+# Release checklist
+
+## Documentation
+- [] Write simplified release notes for FEC with instructions for review
+- [] Update documentation in any relevant repo
+
+### Communciations
+- [] Are there communications around this release from 18F or FEC?
+
+## Reviews
+- [] 18F QA review on staging
+- [] FEC QA review on staging
+- [] File accessibility review request [here](https://github.com/18F/Accessibility_Reviews/issues/new)
+
+## Deployement
+- [] Rebuild tables if necessary
```

Comment:
@noahmanger:  Are you referring to database tables here, like what we had to do with running the update script?  Because that's with the API and not the web app, unless this is meant to serve as a general release checklist for the entire FEC site.